torch/csrc/jit/pickler.cpp

```diff
@@ -151,28 +178,109 @@ void Pickler::pushString(const std::string& string) {
   stack_.insert(stack_.end(), string.begin(), string.end());
 }
 
-void Pickler::pushClass(PicklerClass cls) {
-  const auto& name = getClassName(cls);
-  // Write it to the tensor table
+void Pickler::pushGlobal(const std::string& name) {
   auto memo_entry = memo_map_.find(&name);
```

Comment:
Since the maps are `const` doesn't that imply that there will be no re-allocating / re-hashing and so the pointers will always be valid?

Reply:
Based on how pushGlobal is called, no, it is not always valid. I see pushGlobal being called as:
1. with a const char* string
2. with a string generated from stringstream.
3. with a string using string concat

Version 2 and 3 fail.  This kind of bug happens because the API looks like it does one thing (take a string), but the real API is suppose to take only statically allocated strings. In this case, it is best not to have an API like this in the first place.

Reply:
Regarding this and #20090, an API like this is nice to have so everything doesn't have to be statically spelled out. Adding a reference to pointer IValues and keeping strings around in a table on the pickler should fix this